hibernate-core/src/test/java/org/hibernate/cfg/annotations/schema/SchemaNamingStrategyTest.java

```diff
@@ -0,0 +1,115 @@
+import org.hibernate.query.NativeQueryWithParenthesesTest.Person;
+
+import javax.persistence.EntityManagerFactory;
+import org.hibernate.tool.hbm2ddl.SchemaExport;
+import org.hibernate.tool.hbm2ddl.SchemaExport.Action;
+import org.hibernate.tool.schema.TargetType;
+import org.hibernate.tool.schema.spi.SchemaFilter;
+import org.hibernate.jpa.test.BaseEntityManagerFunctionalTestCase;
+import org.junit.Test;
+
+/**
+ * Test for HHH-12278
+ *
+ * @author Benoit Besson
+ */
+public class SchemaNamingStrategyTest extends BaseEntityManagerFunctionalTestCase {
+
+	@Override
+	public Class[] getAnnotatedClasses() {
+		return new Class[] { Global.class, Dynamic.class };
+	}
+
+	@Test
+	public void testExportSchemaName() throws IOException {
+
+		// instanciate schema name resolver
+		SchemaNameResolver resolver = new SchemaNameResolverImpl();
+
+		// wrap into default provider
+		SchemaNamingDefaultProvider provider = new SchemaNamingDefaultProvider(resolver);
+
+		// register provider to Hibernate core
+		SchemaNamingProviderLocator.setInstance(provider);
+
+		// set dynamic schema name
+		Properties properties = System.getProperties();
+		properties.setProperty(SchemaNameResolverImpl.SCHEMA_DYNAMIC_NAME_PROPERTY_NAME,
+				SchemaNameResolverImpl.getDynamicSchemaName());
+
+		// export entites filtered in the global backage
+		exportSchema(GlobalSchemaFilter.class);
+		// export entites filtered in the global dynamic
+		exportSchema(DynamicSchemaFilter.class);
+
+		// doInJPA( this::entityManagerFactory, entityManager -> {
+		// entityManager.createNativeQuery(
+		//
+		// } );
+	}
+	
+	//copied from superclass as it is not provided as protected
+	private PersistenceUnitDescriptor buildPersistenceUnitDescriptor() {
+		return new TestingPersistenceUnitDescriptorImpl( getClass().getSimpleName() );
+	}
+
+	private void exportSchema(Class<? extends SchemaFilter> schemaFilterProviderClass) throws IOException {
+
+		// set export filter
+		Properties properties = System.getProperties();
+		properties.put("jpa.properties.hibernate.hbm2ddl.schema_filter_provider", schemaFilterProviderClass.getName());
+
+		// configure NamingStrategy
+		{
+			EntityManagerFactoryBuilderImpl builder = (EntityManagerFactoryBuilderImpl) Bootstrap
+					.getEntityManagerFactoryBuilder(buildPersistenceUnitDescriptor(), buildSettings());
+			// extract metadata
+			MetadataImplementor metadata = builder.getMetadata();
+
+			// export schema to check
+			SchemaExport schemaExport = new SchemaExport();
+			schemaExport.setFormat(true);
+			schemaExport.setDelimiter(";");
+			schemaExport.setHaltOnError(true);
+			schemaExport.setImportFiles(null);
+			File outputFile = File.createTempFile("schema-screate", "sql");
+			schemaExport.setOutputFile(outputFile.getAbsolutePath());
+
+			// write to stdout and file
+			schemaExport.execute(EnumSet.of(TargetType.SCRIPT, TargetType.STDOUT), Action.CREATE, metadata);
+
+			// TODO : add assertion for expected file content
```

Comment:
The  asserts are a must have.

Reply:
I agree.

I made this first proposal to discuss on functional usage without making the code too complex to understand. After discussing on how this change could be proposed, we could discuss on how making the good asserts.

Several approches may be done:  testing the generated schema file content, and/or testing metadata or other asserts. Can you help me deciding the good assert(s) to be added ?